File: datasets/python_source-code/Multiple-C1/triple_combo_1.py
"""
Graph BFS - Triple-Combo Attack Submission
Derived from organic_5.py (class-based Graph with bfs()). Stacks three
evasion techniques: identifier renaming, method/statement reordering,
and injected dead code (unused variables and no-op branches that never
affect the output).
"""

from collections import deque


class NetworkGraph:
    def __init__(self):
        self.links = {}
        self._debug_counter = 0  # dead code: never used meaningfully

    def traverse_breadth_first(self, origin):
        # dead code: this flag is set but never actually changes behavior
        cache_enabled = True
        if cache_enabled and False:
            print("cache path never taken")

        visited_set = {origin}
        result_path = []
        pending_queue = deque([origin])

        while pending_queue:
            current_vertex = pending_queue.popleft()
            result_path.append(current_vertex)

            for adjacent in self.links.get(current_vertex, []):
                if adjacent not in visited_set:
                    visited_set.add(adjacent)
                    pending_queue.append(adjacent)

        unused_summary = len(result_path) * 0  # dead code: computed, never used
        return result_path

    def connect(self, node_a, node_b):
        self._debug_counter += 1  # dead code: tracked, never read meaningfully
        self.links.setdefault(node_a, []).append(node_b)
        self.links.setdefault(node_b, []).append(node_a)


if __name__ == "__main__":
    network = NetworkGraph()
    for x, y in [("A", "B"), ("A", "C"), ("B", "D"), ("C", "D"), ("D", "E")]:
        network.connect(x, y)

    print(network.traverse_breadth_first("A"))

File: server/datasets/controlled_benchmarks/java/multiple_files/multiple_c1/triple_combo_1.java
/*
 * Graph BFS - Triple-Combo Attack Submission
 * Derived from organic_5.java. Stacks identifier renaming, method/statement
 * reordering, and injected dead code.
 */

import java.util.*;

class NetworkGraph {
    Map<String, List<String>> links;
    int debugCounter;

    public NetworkGraph() {
        this.links = new HashMap<>();
        this.debugCounter = 0;
    }

    public List<String> traverseBreadthFirst(String origin) {
        boolean cacheEnabled = true;
        if (cacheEnabled && false) {
            System.out.println("cache path never taken");
        }

        Set<String> visitedSet = new HashSet<>();
        List<String> resultPath = new ArrayList<>();
        Queue<String> pendingQueue = new ArrayDeque<>();

        visitedSet.add(origin);
        pendingQueue.add(origin);

        while (!pendingQueue.isEmpty()) {
            String currentVertex = pendingQueue.poll();
            resultPath.add(currentVertex);

            List<String> neighbors = this.links.getOrDefault(currentVertex, Collections.emptyList());
            for (String adjacent : neighbors) {
                if (!visitedSet.contains(adjacent)) {
                    visitedSet.add(adjacent);
                    pendingQueue.add(adjacent);
                }
            }
        }

        int unusedSummary = resultPath.size() * 0;
        return resultPath;
    }

    public void connect(String nodeA, String nodeB) {
        this.debugCounter++;
        this.links.computeIfAbsent(nodeA, k -> new ArrayList<>()).add(nodeB);
        this.links.computeIfAbsent(nodeB, k -> new ArrayList<>()).add(nodeA);
    }

    public static void main(String[] args) {
        NetworkGraph network = new NetworkGraph();
        String[][] pairs = {
            {"A", "B"}, {"A", "C"}, {"B", "D"}, {"C", "D"}, {"D", "E"}
        };
        for (String[] p : pairs) {
            network.connect(p[0], p[1]);
        }
        System.out.println(network.traverseBreadthFirst("A"));
    }
}
